### DISCUSSION TITLE
[HttpClient] Dump failed requests as reproducible (curl?) file

### DISCUSSION THREAD

**Discussion by stephanvierkant** [symfony/symfony#50588] [Q&A]
I'm consuming an external API and because some requests are failing, the other party is asking for details about failing requests (data, headers).

This is what I've done:

```
// \Symfony\Contracts\HttpClient\HttpClientInterface
$response = $this->httpCclient->request(Request::METHOD_POST, $action, [
    'base_uri' => $uri,
    'headers' => [
        'APIKey' =>  'fooBar',
        // ...
    ],
    'json' => $data,
);

if ($response->getStatusCode() !== Response::HTTP_OK) {
    $this->logger->alert('fail!', ['data' => json_encode($data)]);
}
```

It would be great if I could export my request in a format the other party (not PHP developers) could understand. I only found [this question](https://stackoverflow.com/q/75285519/1405981). Any ideas?

**Comment by MatTheCat** [comment]
If you can reproduce the problem on your development environment, the HTTP Client profiler panel should expose a control to copy the request as cURL command.

Have a look at `HttpClientDataCollector::getCurlCommand`.
